PowerPointLabs/PowerPointLabs/AudioMisc/Audio.cs

```diff
@@ -41,6 +45,27 @@ public Audio(string name, string saveName, int matchScriptID)
             Type = AudioHelper.GetAudioType(saveName);
         }
 
+        public Audio(Shape shape, string saveName)
+        {
+            // detect audio type from shape tag
+            AudioType audioType;
+            if (!Enum.TryParse<AudioType>(shape.Tags[AudioTypeTagName], out audioType))
+            {
+                audioType = AudioType.Unrecognized;
+            }
+            this.Type = audioType;
+
+            // derive matched id from shape name
+            var temp = shape.Name.Split(new[] { ' ' });
+            this.MatchScriptID = Int32.Parse(temp[2]);

```

Comment:
This was from RecorderTaskPane.cs, line 436. I can make a check and indicate the format is unrecognized.